src/machine/machine_stm32_spi.go

```diff
@@ -0,0 +1,107 @@
+// +build stm32
+
+package machine
+
+// Peripheral abstraction layer for SPI on the stm32 family
+
+import (
+	"device/stm32"
+	"unsafe"
+)
+
+// SPI phase and polarity configs CPOL and CPHA
+const (
+	SPI_ClkLowValid1stEdge  = 0
```

Comment:
Since these are well known modes do we even need to define them? https://en.wikipedia.org/wiki/Serial_Peripheral_Interface#Mode_numbers

If we do define them, it should probably be in the `spi.go` file, and should be `Mode0` `Mode1` etc.

Reply:
Oh interesting, I wasn't aware there was a widespread definition of those; I just tried to capture what was in the datasheet in a human understandable way. I'll set up the `Mode0`, `..1`, etc in `spi.c`